File: 백준/Gold/7570. 줄 세우기/줄 세우기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;

        int n = Integer.parseInt(br.readLine());
        int[] arr = new int[n + 1];

        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            int num = Integer.parseInt(st.nextToken());
            arr[num] = arr[num - 1] + 1;
        }

        Arrays.sort(arr);
        System.out.println(n - arr[n]);
    }
}

File: 백준/Gold/7570. 줄 세우기/줄 세우기.cc
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
  int n;
  cin >> n;

  int arr[n + 1];
  fill(arr, arr + n + 1, 0);
  for (int i = 0; i < n; i++) {
    int num;
    cin >> num;
    arr[num] = arr[num - 1] + 1;
  }

  sort(arr, arr + n + 1);
  cout << n - arr[n] << endl;
  return 0;
}